pkgs/by-name/ja/jailer/package.nix

```
@@ -0,0 +1,82 @@
+{
+  copyDesktopItems,
+  fetchFromGitHub,
+  lib,
+  makeDesktopItem,
+  makeWrapper,
+  stdenv,
+  wrapGAppsHook4,
+  ant,
+  jdk,
+  jre,
+  stripJavaArchivesHook,
+}:
+stdenv.mkDerivation rec {
+  name = "jailer";
+  version = "16.1.4";
+
+  src = fetchFromGitHub {
+    owner = "Wisser";
+    repo = "Jailer";
+    rev = "dcde48b2e0c7b68cc1b6f73c80fde06fdf31fdc6";
+    sha256 = "sha256-cPOWW2z44vCc/Fw1zYZUsYafgKr82e0g+K0Db5A9F5M=";
+  };
+
+
+  buildPhase = ''
+    runHook preBuild
+    rm jailer.jar
+    ant
+    runHook postBuild
+  '';
+
+  nativeBuildInputs = [ ant jdk stripJavaArchivesHook makeWrapper wrapGAppsHook4 copyDesktopItems ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm644 jailer.jar $out/share/java/jailer.jar
+    install -Dm644 jailer-engine-16.1.4.jar $out/share/java/
+    mkdir -p $out/share/java/lib
+    for f in lib/*.jar; do
+      install -Dm644 $f $out/share/java/lib
+    done
+
+    mkdir -p $out/bin
+    mkdir -p $out/share
+    mkdir -p $out/share/pixmaps
+    cp driverlist.csv $out/share
+    cp admin/jailer.png $out/share/pixmaps
+
+    # On first run, create a local configuration folder and copy driverlist.csv there.
+    cat << EOF > $out/bin/jailer
+    #!/usr/bin/env bash
+    CFG="''${XDG_CONFIG_HOME:-\$HOME/.config}/jailer"
+    mkdir -p \$CFG
+    cp -n $out/share/driverlist.csv \$CFG
+    cd \$CFG
+    _JAVA_AWT_WM_NONREPARENTING=1 ${jre}/bin/java -jar $out/share/java/jailer.jar
+    EOF
+    chmod +x $out/bin/jailer
+
+    runHook postInstall
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "Jailer";
+      desktopName = "Jailer";
+      exec = "jailer";
+      icon = "jailer";
+    })
+  ];
+
+  meta = with lib; {
+    description = "Jailer is a tool for database subsetting and relational data browsing";
+    license = licenses.asl20;
+    homepage = "https://github.com/Wisser/Jailer";
+    changelog = "https://github.com/Wisser/Jailer/releases/tag/${version}";
+    maintainers = with maintainers; [ guillaumematheron ];
+    mainProgram = "jailer";
```

[Avoid `with lib;`](https://github.com/NixOS/nixpkgs/issues/208242)

```suggestion
  meta = {
    description = "Jailer is a tool for database subsetting and relational data browsing";
    license = lib.licenses.asl20;
    homepage = "https://github.com/Wisser/Jailer";
    changelog = "https://github.com/Wisser/Jailer/releases/tag/${finalAttrs.version}";
    maintainers = with lib.maintainers; [ guillaumematheron ];
    mainProgram = "jailer";
```